src/coreclr/jit/gentree.h

```diff
@@ -3298,16 +3301,16 @@ struct GenTreeStrCon : public GenTree
 };
 
 // Common supertype of LCL_VAR, LCL_FLD, REG_VAR, PHI_ARG
-// This inherits from UnOp because lclvar stores are Unops
-struct GenTreeLclVarCommon : public GenTreeUnOp
+// This inherits from GenTreeOp because CSTORE_LCL_VAR has 2 ops
+struct GenTreeLclVarCommon : public GenTreeOp
```

Comment:
This is not a viable approach. It increases the size of small nodes by one pointer (80 -> 88 bytes on 64 bit hosts).

Reply:
> This is not a viable approach. It increases the size of small nodes by one pointer (80 -> 88 bytes on 64 bit hosts).

If that's the case then that also gets rid of my idea of just using STORE_LCL_VAR nodes for conditional stores (STORE_LCL_VAR would then just check in the emit if a conditional flag is set).

Ok then, I think I can just make a new class:
struct GenTreeCondLclVar : public GenTreeLclVarCommon

Reply:
What is the reason for attaching this to local stores at all?

I would expect you can just define a new `GT_SELECT`-like oper, that uses the implicit condition to produce a value.

Reply:
My original instinct was to avoid adding new nodes into the graph, as I was assuming that would result in additional instructions being generated (a csel for the new node, then a mov for the store). But, I'm now realising the register allocator would allocate correctly and the lcl var store would end up as no instruction.

Also, I think my code isn't going to work in cases where the STORE_LCL_VAR needed to store to memory (which probably explains why the c# libraries fail to build)

Reply:
> But, I'm now realising the register allocator would allocate correctly and the lcl var store would end up as no instruction.

Right. "Combining" stores to locals with instructions, if taken to its logical conclusion, would mean adding local stores for all opers, `STORE_LCL_VAR_[ADD|SUB|MUL...]`, so that's not something should do.

(Well -- as a little history detour -- we used to have legacy codegen that did that, but it was mainly motivated by the RMW semantics of most x86 instructions)